corehq/apps/users/forms.py

```diff
@@ -485,18 +485,37 @@ class NewMobileWorkerForm(forms.Form):
         required=False,
         label=ugettext_noop("Last Name")
     )
+    location_id = forms.ChoiceField(
+        label=ugettext_noop("Location"),
+        required=False,
+    )
     password = forms.CharField(
         widget=PasswordInput(),
         required=True,
         min_length=1,
         label=ugettext_noop("Password")
     )
 
-    def __init__(self, domain, *args, **kwargs):
+    def __init__(self, project, *args, **kwargs):
         super(NewMobileWorkerForm, self).__init__(*args, **kwargs)
-        email_string = u"@{}.commcarehq.org".format(domain)
+        email_string = u"@{}.commcarehq.org".format(project.name)
         max_chars_username = 80 - len(email_string)
 
+        if project.uses_locations:
+            locations = SQLLocation.by_domain(project.name)
+            self.fields['location_id'].choices = [(l.location_id, l.name) for l in locations]
```

Comment:
nit: it would likely be more performant to use `values_list` (less IO, less python wrapping)